please/build.go

```diff
@@ -17,31 +17,46 @@ func Build(plz, target string) ([]string, error) {
 
 // RecursivelyProvide queries the target, checking if it provides a different target for the given requirement, and
 // returns that, repeating the operation if that target also provides a different target.
-func RecursivelyProvide(plz, target, requires string) (string, error) {
+func RecursivelyProvide(plz string, target string, requires string) ([]string, error) {
 	out, err := execPlease(plz, "query", "print", "--json", "--field=provides", target)
 	if err != nil {
-		return "", err
+		return []string{}, err
 	}
 	// Returns a map of labels to fields. One of the fields in provides, which is a map of requests to targets, so we
 	// index res[target]["provides"][requires] to get the provided target
-	res := map[string]map[string]map[string]string{}
+	res := map[string]map[string]map[string][]string{}
 	if err := json.Unmarshal(out, &res); err != nil {
-		return "", err
+		return []string{}, err
 	}
 
 	t, ok := res[target]
 	if !ok {
-		return target, nil
+		return []string{target}, nil
 	}
 
 	provides, ok := t["provides"]
 	if !ok {
-		return target, nil
+		return []string{target}, nil
 	}
 
-	ret := provides[requires]
-	if ret == "" || ret == target {
-		return target, nil
+	rets := provides[requires]
+
+	if len(rets) == 0 || (len(rets) == 1 && rets[0] == target) {
+		return []string{target}, nil
+	}
+
+	ret2 := make([]string, 0, len(rets))
```

Comment:
nit: variable names are confusing. Typically we call a value we're building up to return `ret`. We don't return `rets` and there's no `ret` but there's a `ret2`. I might suggest `rets -> providedTargets` and `ret2 -> ret`.

Reply:
Done